src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestNode.cs

```diff
@@ -109,9 +136,22 @@ public HttpBindingResult<HttpRequestMessage> TryGetHttpRequestMessage(HttpBindin
         {
             foreach (var headerNode in headerNodes)
             {
+                if (headerNode.NameNode is null)
+                {
+                    // FIX: (TryGetHttpRequestMessage) add diagnostic
```

Comment:
This seems like an example of a case where calling `TryGetRequest` will report an error if the header name was missing. However, the loop inside `HttpRequestKernel` where we only bind variable expressions will not.

This also seems like a syntax error - so it is odd that we need to do binding to report this error.

Reply:
This is another reason why I feel that we should not try to bind stuff when syntax errors are present under the node (warnings are fine).

If we add an early out for the case where `node.GetDiagnostics()` contains errors, we could throw an exception over here (since we know that the parser would already have reported a diagnostic for this case and if it didn't that is a bug / exceptional condition).

Reply:
> This seems like an example of a case where calling TryGetRequest will report an error if the header name was missing.

This is a syntactic error and a diagnostic should be reported even if binding isn't attempted.

Reply:
I think we should early out if there's a syntactic error in the same node that contains bindings but otherwise, we might as well bind and report every error we know about.

The decision not to proceed with sending a request belongs only at the highest level.

Reply:
My main concern was basically that if we make this a binding-only diagnostic, then we will also need to account for it in other places where we compute and report diagnostics such as over [here]( https://github.com/dotnet/interactive/blob/main/src/Microsoft.DotNet.Interactive.HttpRequest/HttpRequestKernel.cs#L270) - but if we can always report a syntax error for this that sounds great.

> I think we should early out if there's a syntactic error in the same node that contains bindings but otherwise, we might as well bind and report every error we know about.

It may not be a good experience to see 2 errors about the same problem (e.g., header name was missing) if they are reported from both syntax and binding. Thoughts around how we would prevent that?

Reply:
I think I'm not seeing how you would end up with a repeat diagnostic. If there's a missing header value (resulting in a syntactic diagnostic) then there can't be a binding diagnostic because there's nothing to attempt to bind. The binding diagnostics in any case will indicate e.g. an undefined variable.

Reply:
Ah I thought the `TODO` comment above implies that we should report a diagnostic here during binding if `HeaderNameNode` is `null`. Would this diagnostic not be a duplicate of the diagnostic that the parser would add to the tree when it finds that the name is missing?

My assumption is that all syntactic errors should be reported by the parser as it discovers missing / incorrect constructs during parsing. And I assumed you were saying the same thing in this comment -
> This is a syntactic error and a diagnostic should be reported even if binding isn't attempted.

Since `TryGetRequest` currently includes all node level syntax errors under the `HttpRequestNode` as part of the returned diagnostics (on line 97 above), I think we would end up returning 2 diagnostics for this if my understanding is correct. But I am probably missing something obvious / fundamental...

Reply:
I don't think there would be a duplicate but further testing will make it clear if there's an issue.